src/sentry/static/sentry/app/views/eventsV2/tags.tsx

```diff
@@ -66,39 +69,54 @@ class Tags extends React.Component<Props, State> {
   }
 
   shouldRefetchData = (prevProps: Props): boolean => {
-    const thisAPIPayload = this.props.eventView.getTagsAPIPayload(this.props.location);
-    const otherAPIPayload = prevProps.eventView.getTagsAPIPayload(prevProps.location);
+    const thisAPIPayload = pickTagParams(
+      this.props.eventView.getEventsAPIPayload(this.props.location)
+    );
+    const otherAPIPayload = pickTagParams(
+      prevProps.eventView.getEventsAPIPayload(prevProps.location)
+    );
 
-    return !isAPIPayloadSimilar(thisAPIPayload, otherAPIPayload);
+    return !isEqual(thisAPIPayload, otherAPIPayload);
   };
 
   fetchData = async () => {
     const {api, organization, eventView, location} = this.props;
 
-    this.setState({tags: {}, totalValues: null});
+    const tagsFetchID = Symbol('tagsFetchID');
+    this.setState({tags: {}, totalValues: null, tagsFetchID});
```

Comment:
How does `tagsFetchID` get cleared?

Reply:
Due to the nature of the requests being performed to populate the tags; it’s not meant to be cleared. The purpose is to have a unique sentinel value associated with the group of API calls made whenever fetchData has a called.

Reply:
:+1: Makes sense. We should re-fetch tag keys when project lists are updated as the available keys will be different.